crates/workspace/src/toml.rs

```diff
@@ -0,0 +1,117 @@
+//! Utilities for locating (and extracting configuration from) an air.toml.
+
+use crate::toml_options::TomlOptions;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::io;
+use std::path::{Path, PathBuf};
+
+/// Parse an `air.toml` file.
+pub fn parse_air_toml<P: AsRef<Path>>(path: P) -> Result<TomlOptions, ParseTomlError> {
+    let contents = std::fs::read_to_string(path.as_ref())
+        .map_err(|err| ParseTomlError::Read(path.as_ref().to_path_buf(), err))?;
+
+    toml::from_str(&contents)
+        .map_err(|err| ParseTomlError::Deserialize(path.as_ref().to_path_buf(), err))
+}
```

Comment:
This is the entire set of parsing code. It's wild how little it is. Everything is set up at compile time instead!